tests/python_package_test/test_dask.py

```diff
@@ -905,22 +910,18 @@ def test_warns_and_continues_on_unrecognized_tree_learner(client):
     client.close(timeout=CLIENT_CLOSE_TIMEOUT)
 
 
-def test_warns_but_makes_no_changes_for_feature_or_voting_tree_learner(client):
-    X = da.random.random((1e3, 10))
-    y = da.random.random((1e3, 1))
-    for tree_learner in ['feature_parallel', 'voting']:
-        dask_regressor = lgb.DaskLGBMRegressor(
-            client=client,
-            time_out=5,
-            tree_learner=tree_learner,
-            n_estimators=1,
-            num_leaves=2
-        )
-        with pytest.warns(UserWarning, match='Support for tree_learner %s in lightgbm' % tree_learner):
-            dask_regressor = dask_regressor.fit(X, y)
-
-        assert dask_regressor.fitted_
-        assert dask_regressor.get_params()['tree_learner'] == tree_learner
+def test_error_on_feature_parallel_tree_learner(client):
+    X = da.random.random((1_000, 10), chunks=(500, 10))
+    y = da.random.random((1_000, 1), chunks=500)
+    dask_regressor = lgb.DaskLGBMRegressor(
+        client=client,
+        time_out=5,
+        tree_learner='feature_parallel',
+        n_estimators=1,
+        num_leaves=2
+    )
+    with pytest.raises(lgb.basic.LightGBMError):
```

Comment:
```suggestion
    with pytest.raises(lgb.basic.LightGBMError, match="Do not support feature parallel in c api"):
```

Can you please add the specific error message in https://github.com/microsoft/LightGBM/issues/3834#issuecomment-782269920 here? In tests that catch errors, I think it's valuable to check that you got the exact error you expected. Otherwise, unexpected errors introduced by code changes can slip past the tests.

Reply:
looks like this was addressed in https://github.com/microsoft/LightGBM/pull/4088/commits/76b99941a8e8c016be9f375f09fe6d0af8e1108e